### PR TITLE
feat: Migrate to npm@latest

### PR DESCRIPTION
**Pull request by d3viant0ne**
Changes templates to execute via `npm` for all supported node version. Can be updated to install npm@latest for the 4.3 / 6 travis runs if we want.

As this pertains to the contrib loaders & plugins, the proposal is to drop yarn and add an npm lock file which can then be used or not used depending on the developers installed nodejs / npm version with the exception of any dependency updates which will require the lock file to be updates.

I didn't add npm to the engines field ( think it's overkill ). Anyone updating packages will most likely be org-maintainer or admin, in which case requiring npm@5 to be installed is completely reasonable.

Closes webpack-contrib/organization#13

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,24 +7,24 @@ matrix:
   fast_finish: true
   include:
     - os: linux
-      node_js: '7'
-      env: WEBPACK_VERSION="2.2.0" JOB_PART=lint
+      node_js: '8'
+      env: WEBPACK_VERSION="2.6.0" JOB_PART=lint
```
Comment (evilebottnawi) [inline .travis.yml]: Maybe we can add `WEBPACK_VERSION` env? This would make it convenient to update all versions

File: src/tasks/travis.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ module.exports = (config) => {
         include: [
           {
             os: 'linux',
-            node_js: '7',
+            node_js: '8',
```
Comment (sapegin) [inline src/tasks/travis.js]: We probably need a variable for this too ;-)

File: .travis.yml [hunk 3]
```diff
@@ -7,26 +7,26 @@ matrix:
...
+      node_js: '8'
+      env: WEBPACK_VERSION="2.6.0" JOB_PART=coverage
 before_install:
   - nvm --version
   - node --version
 before_script:
   - |-
     if [ "$WEBPACK_VERSION" ]; then
-      yarn add webpack@^$WEBPACK_VERSION
+      npm i -g webpack@^$WEBPACK_VERSION
```
Comment (d3viant0ne) [inline .travis.yml]: `npm i` has changed in the local context, the only common denominator between old and new versions would be a global install.

File: src/tasks/travis.js (hunk 2 above)
Comment (d3viant0ne) [inline src/tasks/travis.js]: Variables covering the entire supported nodejs range ..

```
    "minNode": "4.3",
    "latestNodeLTS": "6",
    "latestNode": "8"
```

File: .travis.yml (hunk 3 above)
Comment (d3viant0ne) [inline .travis.yml]: The behavior for `npm i` has changed in node 8. If we were going to use a mix of versions, global was the only common denominator.

Now that we are using npm@5 for everything, `npm i --no-save` is a viable option

**Comment by d3viant0ne** [comment]
Just need to add the npm install in the before_install quick.

File: src/tasks/travis.js [hunk 4]
```diff
@@ -29,29 +29,35 @@ module.exports = (config) => {
...
             env: jobEnv('coverage'),
           },
         ],
       },
       before_install: [
+        `
+if [[ npm -v != 5* ]]; then
+    npm i -g npm@5;
+fi
+`.trim(),
```
Comment (d3viant0ne) [inline src/tasks/travis.js]: @sapegin @bebraw  - This should handle the npm upgrade for node version !== 8.x

File: .travis.yml [hunk 5]
```diff
@@ -19,8 +19,10 @@ matrix:
       node_js: '8'
       env: WEBPACK_VERSION="2.6.0" JOB_PART=coverage
 before_install:
+  - npm i -g npm@latest
```
Comment (evilebottnawi) [inline .travis.yml]: Very dangerous, the new version sometimes does not work as expected (example npm 5.0.0-5.0.1)

Comment (d3viant0ne) [inline .travis.yml]: I've limited the install range to `semver: Patch`. If it's a choice between occasionally breaking a patch version and having to pin vs. constantly having to update versions in 58 repositories, i'll take the former any day of the week personally.

**Comment by d3viant0ne** [comment]
@sapegin - Ship it

**Pull request merged**